web/modules/custom/dpl_fbi/src/Plugin/Field/FieldType/CqlSearchItem.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\dpl_fbi\Plugin\Field\FieldType;
+
+use Drupal\Component\Utility\Random;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\TypedData\DataDefinition;
+
+/**
+ * Defines the 'dpl_fbi_cql_search' field type.
+ *
+ * @FieldType(
+ *   id = "dpl_fbi_cql_search",
+ *   label = @Translation("CQL Search"),
+ *   category = @Translation("General"),
+ *   default_widget = "string_textfield",
+ *   default_formatter = "string",
+ * )
+ */
+final class CqlSearchItem extends FieldItemBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isEmpty(): bool {
+    return match ($this->get('value')->getValue()) {
+      NULL, '' => TRUE,
+      default => FALSE,
+    };
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function propertyDefinitions(FieldStorageDefinitionInterface $field_definition): array {
+
+    // @DCG
+    // See /core/lib/Drupal/Core/TypedData/Plugin/DataType directory for
+    // available data types.
+    $properties['value'] = DataDefinition::create('string')
+      ->setLabel(t('Text value'))
+      ->setRequired(TRUE);
+
+    return $properties;
+  }
+
+  /**
+   * Getting the constraints for the field.
+   *
+   * @return array<mixed>
```

Comment:
:+1: I see that you resolved the static analysis errors whatever they were without resolving to a global ignore.